### PR TITLE
[Merged by Bors] - feat(Analysis/Complex/ReImTopology): Uniform continuity of re im

### PR DESCRIPTION
**Pull request by CBirkbeck**
We show that re and im are uniformly continuous and then use this to show that re/im parts of a uniform limit converge uniformly.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/Complex/Basic.lean [hunk 1]
```diff
@@ -241,6 +241,12 @@ def reCLM : ℂ →L[ℝ] ℝ :=
 theorem continuous_re : Continuous re :=
   reCLM.continuous
 
+lemma UniformlyContinous_re : UniformContinuous re := by
+  rw [Metric.uniformContinuous_iff]
+  intro ε hε
+  refine ⟨ε, hε, fun hxy => ?_⟩
+  apply lt_of_le_of_lt (Complex.abs_re_le_abs _) hxy
```
Comment (urkud) [inline Mathlib/Analysis/Complex/Basic.lean:248]:
```suggestion
lemma uniformContinous_re : UniformContinuous re :=
  reCLM.uniformContinuous
```
same below

File: Mathlib/Analysis/Complex/ReImTopology.lean [hunk 2]
```diff
@@ -175,3 +175,19 @@ theorem IsClosed.reProdIm (hs : IsClosed s) (ht : IsClosed t) : IsClosed (s ×
 
 theorem Bornology.IsBounded.reProdIm (hs : IsBounded s) (ht : IsBounded t) : IsBounded (s ×ℂ t) :=
   antilipschitz_equivRealProd.isBounded_preimage (hs.prod ht)
+
+section continuity
+
+variable {α ι : Type*}
+
+lemma TendstoUniformlyOn_re_part (f : ι → α → ℂ) {p : Filter ι} (g : α → ℂ) (K : Set α)
+    (hf : TendstoUniformlyOn f g p K) : TendstoUniformlyOn (fun n x => (f n x).re)
```
Comment (urkud) [inline Mathlib/Analysis/Complex/ReImTopology.lean:184]:
```suggestion
protected lemma TendstoUniformlyOn.re {f : ι → α → ℂ} {p : Filter ι} {g : α → ℂ} {K : Set α}
    (hf : TendstoUniformlyOn f g p K) : TendstoUniformlyOn (fun n x => (f n x).re)
```
same below

Comment (urkud) [reply Mathlib/Analysis/Complex/ReImTopology.lean:184]: Also, please add a version for `TendstoUniformly`.

File: Mathlib/Analysis/Complex/ReImTopology.lean [hunk 3]
```diff
@@ -175,3 +175,29 @@ theorem IsClosed.reProdIm (hs : IsClosed s) (ht : IsClosed t) : IsClosed (s ×
...
+      (fun y => (g y).re) p K := by
+  apply UniformContinuous.comp_tendstoUniformlyOn UniformlyContinous_re hf
+
+protected lemma TendstoUniformly.re {f : ι → α → ℂ} {p : Filter ι} {g : α → ℂ}
+    (hf : TendstoUniformly f g p) : TendstoUniformly (fun n x => (f n x).re)
+      (fun y => (g y).re) p := by
+  apply UniformContinuous.comp_tendstoUniformly UniformlyContinous_re hf
+
+protected lemma TendstoUniformlyOn.im (f : ι → α → ℂ) {p : Filter ι} {g : α → ℂ} {K : Set α}
```
Comment (urkud) [inline Mathlib/Analysis/Complex/ReImTopology.lean:193]: `f` can be an implicit argument here and below.

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**